File: app/src/main/java/com/ddmyb/shalendar/RegisterActivity.java
package com.ddmyb.shalendar;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class RegisterActivity extends AppCompatActivity {

    private FirebaseAuth mFirebaseAuth; //파이어베이스 인증 처리
    private DatabaseReference mDatabaseRef; //실시간 데이터베이스
    private EditText mEtEmail, mEtPwd; //회원가입 입력칸
    private Button mBtnRegister;    //회원가입 버튼
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        mFirebaseAuth = FirebaseAuth.getInstance(); // 파이어베이스 데이터베이스 연동
        mDatabaseRef = FirebaseDatabase.getInstance().getReference("MaengDol");

        mEtEmail = findViewById(R.id.et_email);
        mEtPwd = findViewById(R.id.et_pwd);
        mBtnRegister = findViewById(R.id.btn_register);

        mBtnRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //회원가입 처리 시작
                String strEmail = mEtEmail.getText().toString();
                String strPwd = mEtPwd.getText().toString();
                Toast.makeText(RegisterActivity.this, "진행중", Toast.LENGTH_SHORT).show();
                //Firebase Auth 진행
                mFirebaseAuth.createUserWithEmailAndPassword(strEmail,strPwd).addOnCompleteListener(RegisterActivity.this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if(task.isSuccessful()) {
                            FirebaseUser firebaseUser = mFirebaseAuth.getCurrentUser();

                            UserAccount account = new UserAccount();
                            account.idToken = (firebaseUser.getUid());
                            account.emailId = firebaseUser.getEmail();
                            account.password = strPwd;

                            Toast.makeText(RegisterActivity.this, "진행중4", Toast.LENGTH_SHORT).show();
                            //child는 해당 키 위치로 이동하는 함수입니다.
                            //키가 없는데 "UserAccount"와 firebaseUser.getUid()같이 값을 지정한 경우 자동으로 생성합니다.
                            mDatabaseRef.child("UserAccount").child(firebaseUser.getUid()).setValue(account);

                            Toast.makeText(RegisterActivity.this, "회원가입에 성공하셨습니다", Toast.LENGTH_LONG).show();
                        } else {
                            Toast.makeText(RegisterActivity.this, "회원가입에 실패하셨습니다", Toast.LENGTH_LONG).show();
                        }
                    }
                });
            }
        });

    }
}

File: app/src/main/java/com/ddmyb/shalendar/RegisterActivity.kt
package com.ddmyb.shalendar

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.ddmyb.shalendar.databinding.ActivityLoginBinding
import com.ddmyb.shalendar.databinding.ActivityRegisterBinding
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class RegisterActivity : AppCompatActivity() {
    private var mFirebaseAuth: FirebaseAuth? = null //파이어베이스 인증 처리
    private var mDatabaseRef: DatabaseReference? = null //실시간 데이터베이스

    private val binding by lazy {
        ActivityRegisterBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        mFirebaseAuth = FirebaseAuth.getInstance() // 파이어베이스 데이터베이스 연동
        mDatabaseRef = FirebaseDatabase.getInstance().getReference("UsersAccount")

        binding.btnRegister.setOnClickListener(View.OnClickListener {
            //회원가입 처리 시작
            val strNickname = binding.etNickname.getText().toString()
            val strEmail = binding.etEmail.getText().toString()
            val strPwd = binding.etPwd.getText().toString()
            Toast.makeText(this@RegisterActivity, "진행중", Toast.LENGTH_SHORT).show()
            //Firebase Auth 진행
            mFirebaseAuth!!.createUserWithEmailAndPassword(strEmail, strPwd).addOnCompleteListener(this@RegisterActivity)
            { task ->
                if (task.isSuccessful) {
                    val firebaseUser = mFirebaseAuth!!.currentUser
                    val account = UserAccount().apply {
                        idToken = firebaseUser!!.uid
                        emailId = firebaseUser.email
                        nickName = strNickname
                        password = strPwd
                    }
                    Toast.makeText(this@RegisterActivity, "진행중", Toast.LENGTH_SHORT).show()
                    //child는 해당 키 위치로 이동하는 함수입니다.
                    //키가 없는데 "UserAccount"와 firebaseUser.getUid()같이 값을 지정한 경우 자동으로 생성합니다.
                    mDatabaseRef!!.child(firebaseUser!!.uid).setValue(account)
                    //mDatabaseRef!!.child(firebaseUser!!.email.toString().replace(".","_")).setValue(account)
                    Toast.makeText(this@RegisterActivity, "회원가입에 성공하셨습니다", Toast.LENGTH_LONG).show()
                } else {
                    Toast.makeText(this@RegisterActivity, "회원가입에 실패하셨습니다", Toast.LENGTH_LONG).show()
                }
            }
        })
    }
}